src/components/molecules/Message/Message.jsx

```diff
@@ -1,55 +1,45 @@
 import PropTypes from 'prop-types';
-import { useCallback } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { ImageComponent, Text } from '../..';
 import { routes } from '../../../routes/Routes';
-import chatAvatar from '../../../images/avatar.png';
 import { dateFormat } from '../../../utils/format';
 
 import './Message.scss';
+import { useUserDetails } from '../../../hooks';
 
-export const MessageComponent = ({
-    children,
-    date,
-    name,
-    avatar,
-    isYours,
-    profileID
-}) => {
+export const MessageComponent = ({ children, date, isYours, profileID }) => {
     const redirect = useNavigate();
+    const userDetails = useUserDetails(profileID);
 
     return (
-        <div className={`message__field ${isYours ? 'ownMessage' : ''}`}>
-            <ImageComponent
-                src={avatar}
-                size='small'
-                onClick={useCallback(() => {
-                    redirect(`${routes.Feed}${routes.Profile}/${profileID}`);
-                }, [profileID, redirect])}
-            />
-            <Text size='small' customClass='name'>
-                {name}
-            </Text>
-            <Text size='small' customClass='date'>
-                {dateFormat(date)}
-            </Text>
-            <Text type='primary' customClass='message'>
-                {children}
-            </Text>
-        </div>
+        userDetails.isSuccess && (
+            <div className={`message__field ${isYours ? 'ownMessage' : ''}`}>
+                <ImageComponent
+                    src={userDetails.data.data().profile_picture_url}
+                    size='small'
+                    onClick={() => redirect(`${routes.Profile}/${profileID}`)}
```

Comment:
generally here, we should close the userDetails in useMemo, I'm not sure, but components could reload many times where we receive new messages, so we don't want to rerender every messages, also put the onClick function to useCallback, 
This particular component is great example of component that could be pure (also it shouldn't be responsible for API call, but as I said before, leave it as it is imo)

Reply:
I don't think useMemo is useful because react query is responsible for caching fetched data and not over fetching it.